ttnn/cpp/ttnn/operations/ccl/all_gather/device/multi_core/all_gather_op_multi_core.cpp

```diff
@@ -885,13 +921,23 @@ operation::ProgramWithCallbacks all_gather_multi_core_with_workers(const Tensor&
                             static_cast<uint32_t>(sender_worker_reader_semaphore_addr),
                             static_cast<uint32_t>(is_clockwise_direction ? 1 : 0),
                             static_cast<uint32_t>(cb_num_pages / 2),
-                            static_cast<uint32_t>(ring_size)
+                            static_cast<uint32_t>(ring_size),
+                            static_cast<bool>(fuse_op)
                         };
 
                         if (is_sharded) {
                             emit_sharded_tensor_kernel_ct_args(device, output_tensor, worker_writer_receiver_ct_args, output_pages_per_shard_y, output_pages_per_shard_x);
                         }
 
+                        if (fuse_op) {
+                            all_gather_fused_op_signaler->emit_all_gather_fused_op_ct_args(worker_writer_receiver_ct_args, global_num_workers, b);
+                        } else {
+                            // Push dummy args so that kernel doesn't error out at compile time from the lack of args when fuse_op=false
+                            for (uint32_t w = 0; w < 3; ++w) {
```

Comment:
this hardcoded 3 should be queryable off of AllGatherFusedOpSignaller as a class static function.  Or even better, it can do this padding on its end